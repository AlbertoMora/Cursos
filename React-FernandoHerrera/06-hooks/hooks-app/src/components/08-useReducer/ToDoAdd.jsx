import React from "react";
import { useForm } from "../../hooks/useForm";

const ToDoAdd = ({handleAddTodo}) => {

    const [{description}, handleInputChange, reset] = useForm({ description: "" });

    const handleSubmit = (e) => {
        e.preventDefault();

        if (description.trim().length <= 1) {
            return;
        }

        handleAddTodo({
            id: new Date().getTime(),
            desc: description,
            done: false
        })

        reset();
    };

    return (
        <>
            <h4>Agregar toDo</h4>
            <hr />
            <form onSubmit={handleSubmit}>
                <input
                    className="form-control"
                    type="text"
                    name="description"
                    value={description}
                    onChange={handleInputChange}
                    placeholder="Aprender..."
                    autoComplete="off"
                />
                <button type="submit" className="btn btn-outline-info mt-1 btn-block">
                    +
                </button>
            </form>
        </>
    );
};

export default ToDoAdd;
