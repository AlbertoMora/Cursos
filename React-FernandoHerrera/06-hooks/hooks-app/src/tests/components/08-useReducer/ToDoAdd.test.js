import React from "react";
import { shallow } from "enzyme";
import ToDoAdd from "../../../components/08-useReducer/ToDoAdd";

describe("Tests on <ToDoAdd />", () => {
    const handleAddToDo = jest.fn();
    const wrapper = shallow(<ToDoAdd handleAddTodo={handleAddToDo} />);

    test("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
    test("should not call AddToDo handler", () => {
        const formSubmit = wrapper.find("form").prop("onSubmit");
        formSubmit({ preventDefault() {} });

        expect(handleAddToDo).toBeCalledTimes(0);
    });

    test("should call AddToDo handler", () => {
        const value = "Aprender React";
        wrapper.find("input").simulate("change", {
            target: {
                value,
                name: "description",
            },
        });

        const formSubmit = wrapper.find("form").prop("onSubmit");
        formSubmit({ preventDefault() {} });

        expect(handleAddToDo).toBeCalled();
        expect(handleAddToDo).toBeCalledWith({
            id: expect.any(Number),
            desc: value,
            done: false,
        });

        expect(wrapper.find('input').prop('value')).toBe("");
    });
});
