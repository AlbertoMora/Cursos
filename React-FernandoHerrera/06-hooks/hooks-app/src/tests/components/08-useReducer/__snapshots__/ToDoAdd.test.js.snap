// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <ToDoAdd /> should render correctly 1`] = `
<Fragment>
  <h4>
    Agregar toDo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-info mt-1 btn-block"
      type="submit"
    >
      +
    </button>
  </form>
</Fragment>
`;
